src/ECL/ECL.hs

```diff
@@ -43,15 +62,32 @@ data Guard = CatchAll | Opaque
 -- a collection of values, and an optional guard.
 type Alternative lit = (Binders lit, Maybe Guard)
 
--- | A list of uncovered cases
-newtype Uncovered lit = Uncovered { getUncovered :: [Binders lit] }
-  deriving (Show, Eq) 
+-- | Check wraps both uncovered and redundant cases.
+data Check lit = Check
+  { getUncovered :: [Binders lit]
+  , getRedundant :: (Maybe Bool, [Binders lit])
```

Comment:
@paf31 How should we encode that constraint in the type? Using GADTs? I thought something like `Maybe [Binders lit]` for redundant cases as I explain above.